solr/solrj/src/java/org/apache/solr/client/solrj/io/Tuple.java

```diff
@@ -77,38 +77,26 @@ public Tuple() {
    * A copy constructor.
    * @param fields map containing keys and values to be copied to this tuple
    */
-  public Tuple(Map<?, ?> fields) {
-    for (Map.Entry<?, ?> entry : fields.entrySet()) {
-      put(entry.getKey(), entry.getValue());
-    }
+  public Tuple(Map<String, ?> fields) {
+    this.fields.putAll(fields);
+    EOF = this.fields.containsKey(StreamParams.EOF);
+    EXCEPTION = this.fields.containsKey(StreamParams.EXCEPTION);
```

Comment:
Am not 100% sure re: this refactor in the scenario where a deriving class had overridden the `put` method, though haven't fully researched it and might be confusing C++ and Java logic here too. Plus `put` method overriding seems unlikely.

Reply:
You're right that this might break if somebody had overridden `put`. There were no instances of this happening in our code, but users could have different ideas with plugins. Maybe we should declare `put` as `final`? I can back this change out and leave a note for a future JIRA if you'd prefer.

Reply:
> ... I can back this change out and leave a note for a future JIRA if you'd prefer.

That would be my preference, yes.